src/common/asb/Asb.c

```diff
@@ -2581,36 +2076,310 @@ static char* AuditEnsureRloginServiceIsDisabled(void* log)
+static int InitEnsureSshMaxAuthTriesIsSet(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshMaxAuthTriesIsSetObject, value, log);
+}
+
+static int InitEnsureAllowUsersIsConfigured(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureAllowUsersIsConfiguredObject, value, log);
+}
+
+static int InitEnsureDenyUsersIsConfigured(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureDenyUsersIsConfiguredObject, value, log);
+}
+
+static int InitEnsureAllowGroupsIsConfigured(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureAllowGroupsIsConfiguredObject, value, log);
+}
+
+static int InitEnsureDenyGroupsConfigured(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureDenyGroupsConfiguredObject, value, log);
+}
+
+static int InitEnsureSshHostbasedAuthenticationIsDisabled(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshHostbasedAuthenticationIsDisabledObject, value, log);
+}
+
+static int InitEnsureSshPermitRootLoginIsDisabled(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshPermitRootLoginIsDisabledObject, value, log);
+}
+
+static int InitEnsureSshPermitEmptyPasswordsIsDisabled(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshPermitEmptyPasswordsIsDisabledObject, value, log);
+}
+
+static int InitEnsureSshClientIntervalCountMaxIsConfigured(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshClientIntervalCountMaxIsConfiguredObject, value, log);
+}
+
+static int InitEnsureSshClientAliveIntervalIsConfigured(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshClientAliveIntervalIsConfiguredObject, value, log);
+}
+
+static int InitEnsureSshLoginGraceTimeIsSet(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshLoginGraceTimeIsSetObject, value, log);
+}
+
+static int InitEnsureOnlyApprovedMacAlgorithmsAreUsed(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureOnlyApprovedMacAlgorithmsAreUsedObject, value, log);
+}
+
+static int InitEnsureSshWarningBannerIsEnabled(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureSshWarningBannerIsEnabledObject, value, log);
+}
+
+static int InitEnsureUsersCannotSetSshEnvironmentOptions(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureUsersCannotSetSshEnvironmentOptionsObject, value, log);
+}
+
+static int InitEnsureAppropriateCiphersForSsh(char* value, void* log)
+{
+    return InitializeSshAuditCheck(g_initEnsureAppropriateCiphersForSshObject, value, log);
+}
+
+static int ReplaceString(char* target, char* source, const char* defaultValue)
+{
+    bool isValidValue = ((NULL == source) || (0 == source[0])) ? false : true;
+    int status = 0;
+
+    FREE_MEMORY(target);
+    status = (NULL != (target = DuplicateString(isValidValue ? source : defaultValue))) ? 0 : ENOMEM;

```

Comment:
Is this working as intended? How is the new reference from `DuplicateString` returned back to be deallocated later?

Reply:
Yes, it works, because how the g_desired globals are defined. But I see why the question. Stay tuned for an update.